File: react-native/src/tmdb_api/model/Video.ts
export class MovieVideo {
    id: string = "";
    iso_639_1: string = "";
    key: string = "";
    name: string = "";
    site: string = "";
    size: number = 0;
    type: string = "";

    public toString(): string {
        return `{id: ${this.id}, name: "${this.name}", type: ${this.type}}`;
    }
}

File: android/model/src/main/java/com/tikal/tmdb/json/model/Video.kt
package com.tikal.tmdb.json.model

import java.util.Locale
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MovieVideo(
    @SerialName("id")
    val id: String,
    @SerialName("iso_639_1")
    private val localeCountry: String? = null,
    @SerialName("iso_3166_1")
    private val localeLanguage: String? = null,
    @SerialName("key")
    val key: String?,
    @SerialName("name")
    val name: String?,
    @SerialName("site")
    val site: String?,
    @SerialName("size")
    val size: Int = 0,
    @SerialName("type")
    val type: String?,
) {
    val locale: Locale by lazy { Locale(localeLanguage.orEmpty(), localeCountry.orEmpty()) }
}
